### PR TITLE
feat(explorer): add operators and restakings

### LINKED ISSUE #597: feat(explorer): add operators screen
# Objective

This screen should include data regarding each validators':

* Name
* Stake
* Voting power
* etc.

image

## Useful commands

You can get the total stake with

```
cast call  0x51462D5511563A0F97Bb3Ce5475E1c3905b83F4b \
        "getCurrentTotalStake(uint8 quorumNumber)(uint96)" \
        0 \
        --rpc-url https://ethereum-holesky-rpc.publicnode.com/
```

If you want to check Luganodes stake

```
cast call 0x51462D5511563A0F97Bb3Ce5475E1c3905b83F4b \
	"weightOfOperatorForQuorum(uint8 quorumNumber, address operator) (uint96)" \
	0 0xbb5614cda99012c473b0aaa22904a270d50a149a \
	--rpc-url https://ethereum-holesky-rpc.publicnode.com/
```

For example, another operator, PierTwo

```
cast call 0x51462D5511563A0F97Bb3Ce5475E1c3905b83F4b \
	"weightOfOperatorForQuorum(uint8 quorumNumber, address operator) (uint96)" \
	0 0x4a3d6e0ff640f2f0c1091878e6ed3be7ed3f6665 \
	--rpc-url https://ethereum-holesky-rpc.publicnode.com/
```

### PR DESCRIPTION
**Pull request by glpecile**
> [!NOTE]  
> This PR is currently deployed in [STAGE](https://stage.explorer.alignedlayer.com/) and [PROD](https://explorer.alignedlayer.com)

# Changes

> [!CAUTION]  
> [pnpm](https://pnpm.io/installation) is now a requirement for running the explorer
> Run `make explorer_fetch_old_operators_strategies_restakes` to show operators and restakings data

* ~~merge after #617~~ 
* [feat] add views to see existing aligned operators in table format with data from the various eigenlayer contracts
* [feat] add detail view to see operator data in detail with restaked assets
* [feat] add view to see assets restaked
* [feat] add detail view to see restaked asset data in detail
* [feat] a tooltip js library was added that needs pnpm, the step is added in the new makefile so pnpm is now a requirement for the explorer to run
* [style] add css animations to UI elements improving the look and feel with motion 
* [fix] ensure socket is connected before subscribing to pubsub in order to avoid multiple connections
* [fix] remove @alignedlayer from navbar, add docs link instead
* [feat] add hamburger menu for mobile navbar

## 🆕 Scripts

This script should be run in order for this PR to work.

```
make explorer_fetch_old_operators_strategies_restakes
```

## New tables added

Read more about the EigenLayer architecture [here](https://www.blog.eigenlayer.xyz/ycie/).

image

## New Views

image

image

image

image

image

### CODE REVIEW

**Review by uri-99 — APPROVED** [review]

File: explorer/lib/explorer/contract_managers/delegation_manager.ex [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    abi_file: "lib/abi/DelegationManager.json",
+    default_address: @delegation_manager_address
+
+  def get_delegation_manager_address() do
+    @delegation_manager_address
+  end
+
+  def get_operator_url(operator_address) do
+    DelegationManager.EventFilters.operator_metadata_uri_updated(operator_address)
+      |> Ethers.get_logs(fromBlock: 0)
```
Comment (NicolasRampoldi) [inline explorer/lib/explorer/contract_managers/delegation_manager.ex]: Maybe we can start it in a higher number, I don't think it's necessary to parse every block

File: explorer/lib/explorer/contract_managers/delegation_manager.ex [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          IO.inspect("Error getting operator url")
+          IO.inspect(reason)
+
+        other ->
+          IO.inspect("Unexpected response:")
+          IO.inspect(other)
+      end
+  end
+
+  # this gets of all eigenlayer, not avs. sad
```
Comment (NicolasRampoldi) [inline explorer/lib/explorer/contract_managers/delegation_manager.ex]: Delete comment

**Review by NicolasRampoldi — CHANGES_REQUESTED** [review]

**Review by NicolasRampoldi — APPROVED** [review]

File: explorer/lib/explorer/contract_managers/aligned_layer_service_manager.ex [hunk 3]
```diff
@@ -236,4 +236,14 @@ defmodule AlignedLayerServiceManager do
       :error -> raise("Error parsing @gas_per_proof")
     end
   end
+
+  def update_restakeable_strategies() do
+    case AlignedLayerServiceManager.get_restakeable_strategies() |> Ethers.call() do
+      {:ok, restakeable_strategies} ->
+        Strategies.update(restakeable_strategies)
+      {:error, error} ->
+        dbg("Error fetching restakeable stratergies: #{error}")
```
Comment (entropidelic) [inline explorer/lib/explorer/contract_managers/aligned_layer_service_manager.ex]:
```suggestion
        dbg("Error fetching restakeable strategies: #{error}")
```

File: explorer/lib/explorer/contract_managers/aligned_layer_service_manager.ex [hunk 4]
```diff
@@ -236,4 +236,14 @@ defmodule AlignedLayerServiceManager do
...
     end
   end
+
+  def update_restakeable_strategies() do
+    case AlignedLayerServiceManager.get_restakeable_strategies() |> Ethers.call() do
+      {:ok, restakeable_strategies} ->
+        Strategies.update(restakeable_strategies)
+      {:error, error} ->
+        dbg("Error fetching restakeable strategies: #{error}")
+        raise("Error fetching restakeable stratergies: #{error}")
```
Comment (entropidelic) [inline explorer/lib/explorer/contract_managers/aligned_layer_service_manager.ex]:
```suggestion
        raise("Error fetching restakeable strategies: #{error}")
```

File: explorer/lib/explorer/contract_managers/stake_registry_manager.ex [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          acc
+      end
+    end)
+
+    strategies |> Enum.reverse()
+  end
+
+  def get_stake_of_quorum_for_operator(%Restakings{operator_address: operator_address, quorum_number: nil}) do # AT THE MOMENT, ONLY USING QUORUM 0
+    get_stake_of_quorum_for_operator(%Restakings{operator_address: operator_address, quorum_number: 0})
+  end
```
Comment (entropidelic) [inline explorer/lib/explorer/contract_managers/stake_registry_manager.ex:77]:
```suggestion
  end
  
```

**Review by entropidelic — APPROVED** [review]

**Review by uri-99 — CHANGES_REQUESTED** [review]
not yet

**Review by uri-99 — APPROVED** [review]

**Pull request merged**